File: Data Structures & Algorithms/duplicate-integer/submission-8.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        for (let i = 0; i <= nums.length -1;i++){
            for (let x = i+1; x<= nums.length-1; x++){
                if(nums[i] === nums[x]){
                    return true
                }
            }
        }
        return false
    }

}

File: Data Structures & Algorithms/maximum-product-subarray/submission-8.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res = 0
        curMin, curMax = 1,1
        for num in nums:
            temp = num * curMax
            curMax = max(num, num * curMax, num*curMin)
            curMin = min(num, temp, num*curMin)
            res = max(res, curMax)
        return res
        